### PR TITLE
Sqlite specification, fixes and tests

### LINKED ISSUE #9700: sqlite is stuck in transaction mode after performing one transaction
If you begin a transaction with sqlite and then end it with `commit` or `rollback` you will be placed into _another_ transaction instead of just ending the current and being back in autocommit mode.

The [source](https://github.com/HaxeFoundation/haxe/blob/development/std/cpp/_std/sys/db/Sqlite.hx#L86) reads as follows:
```haxe
public function commit() {
	request("COMMIT");
	startTransaction(); // match mysql usage
}

public function rollback() {
	request("ROLLBACK");
	startTransaction(); // match mysql usage
}
```

I'm not sure what the meaning of "match mysql usage" is here;
Mysql doesn't put you back into another transaction after you end one

The `startTransaction()` lines seem like bugs
Do we know why they are there?

You don't want to always be in a transaction with sqlite. This breaks auto-commit once you've performed one transaction and means further changes aren't written to disk unless you perform another transaction after.
(While with mysql they are, as it does not have this behaviour)

### PR DESCRIPTION
**Pull request by RealyUniqueName**
Fixes #9700 

Adds specification according to neko implementation (I think it was the first one).

Discovered problems:
* cpp: missing implementation for `sys.db.Sqlite`
* hl: runtime exception `src/module.c(352) : FATAL ERROR : Failed to load library sqlite.hdll` which I didn't try to solve
* cs, java: `sys.db.Sqlite` is far from finished
* python, lua: `sys.db.Sqlite` is not implemented

### CODE REVIEW

**Comment by jonasmalacofilho** [comment]
The new `commit()` and `rollback()` behaviors are objectively better, but also breaking changes, particularly for C++ and Neko.

But many probably don't use explicit transactions, and others may already do manual `"COMMIT"` and `"ROLLBACK"` requests, so it's hard to know whether breaking this weird behavior is a problem or not.

---

By the way, [auto spawning transactions were originally introduced in Haxe 1.17](https://github.com/HaxeFoundation/haxe/commit/e2f127ac55648a9fdf81c827bcdb6fb65f3fa069).  I remember seeing Nicolas talking about this strange behavior a few years later, but that discussion has apparently been lost to the sands of time.

File: tests/sys/src/db/SqliteSetup.hx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package db;
+
+import sys.db.Sqlite;
+import sys.db.Connection;
+import sys.FileSystem;
+
+class SqliteSetup extends utest.Test {
+	static public final DB_FILE = 'temp/db.sqlite';
```
Comment (jonasmalacofilho) [inline tests/sys/src/db/SqliteSetup.hx]: Suggestion: use an in-memory database (`DB_FILE = ':memory:'`) and avoid poluting the filesystem and having to manually delete the file and table for every test.

Comment (RealyUniqueName) [reply tests/sys/src/db/SqliteSetup.hx]: Unfortunately `:memory:` drops the data once connection is closed. That makes it impossible to test #9700

Comment (RealyUniqueName) [reply tests/sys/src/db/SqliteSetup.hx]: Changed to in-memory except for that test.

Comment (jonasmalacofilho) [reply tests/sys/src/db/SqliteSetup.hx]: Oh, right, I missed that.

**Comment by ConstNW** [comment]
> hl: runtime exception src/module.c(352) : FATAL ERROR : Failed to load library sqlite.hdll which I didn't try to solve

https://github.com/HaxeFoundation/hashlink/pull/323

**Pull request merged**